File: Week 2 - Weather CLI/Project1-weather-cli/week2/weather_one_city.py
# week2/weather_one_city.py
from pathlib import Path
from dotenv import load_dotenv
import os, requests

# Load the .env that lives in the SAME folder as this file
load_dotenv(Path(__file__).with_name(".env"))

KEY = os.getenv("OPENWEATHER_API_KEY") or exit("Missing OPENWEATHER_API_KEY in week2/.env")

city = "Sao Paulo"
url = f"https://api.openweathermap.org/data/2.5/weather?q={city}&appid={KEY}&units=imperial"
r = requests.get(url, timeout=20)
d = r.json()
print(f"{city}: {d['main']['temp']}°C, Humidity {d['main']['humidity']}%")

File: Week 3 - Node.JS/weather_one_city.js
import dotenv from "dotenv";
import { fileURLToPath } from "url";
import { dirname, join } from "path";

// Load .env next to this script
const __filename = fileURLToPath(import.meta.url);
const __dirname = dirname(__filename);
dotenv.config({ path: join(__dirname, ".env") });

const key = process.env.OPENWEATHER_API_KEY;
if (!key) {
  console.error("Missing OPENWEATHER_API_KEY in .env");
  process.exit(1);
}

// Usage: node weather_one_city.js "Seattle" [metric|imperial]
const city = process.argv[2] || "Seattle";
const units = (process.argv[3] === "imperial") ? "imperial" : "metric";

const url = `https://api.openweathermap.org/data/2.5/weather?q=${encodeURIComponent(city)}&appid=${key}&units=${units}`;

try {
  const res = await fetch(url);
  if (!res.ok) {
    console.error("API error:", res.status, await res.text());
    process.exit(1);
  }
  const data = await res.json();
  const temp = data?.main?.temp;
  const hum = data?.main?.humidity;

  if (temp == null || hum == null) {
    console.error("Unexpected response shape:", JSON.stringify(data, null, 2));
    process.exit(1);
  }

  console.log(`${city}: ${temp}°${units === "metric" ? "C" : "F"}, Humidity ${hum}%`);
} catch (err) {
  console.error("Network or parsing error:", err.message);
  process.exit(1);
}
